modules/local/kraken2/main.nf

```diff
@@ -0,0 +1,39 @@
+process KRAKEN2 {
+    tag "$meta.id"
+    label 'process_medium'
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-5799ab18b5fc681e75923b2450abaa969907ec98:87fc08d11968d081f3e8a37131c1f1f6715b6542-0' :
+        'biocontainers/mulled-v2-5799ab18b5fc681e75923b2450abaa969907ec98:87fc08d11968d081f3e8a37131c1f1f6715b6542-0' }"
+
+    input:
+    tuple val(meta), path(reads), path(kraken2_db)
```

Comment:
I'm wondering if you could pass the database as a separate entry in the `input` section? That way you don't have to combine the database into a channel with the reads, you can pass it separately as it's own channel like:

```nextflow
KRAKEN2 (
    FASTP_TRIM.out.reads,
    ch_kraken2_db
)
```

Reply:
😄 Fixed in [b639792](https://github.com/phac-nml/speciesabundance/pull/5/commits/b63979290f57184c8296aee74fb30fe6239fc47a)